tests/hvm_cpu.rs

```diff
@@ -0,0 +1,399 @@
+use hvm_lang::term::{parser, DefNames, DefinitionBook, Term};
+use hvmc::{ast::*, *};
+use insta::{assert_debug_snapshot, assert_snapshot};
+use std::fs;
+
+// Loads file and generate net from hvm-core syntax
+fn load_from_core(file: &str, size: usize) -> (run::Book, run::Net) {
+  let path = format!("{}/tests/programs/{}", env!("CARGO_MANIFEST_DIR"), file);
+  let code = fs::read_to_string(path).unwrap();
+
+  let book = book_to_runtime(&do_parse_book(&code));
+  let mut net = run::Net::new(size);
+  net.boot(name_to_val("main"));
+  (book, net)
+}
+
+// Loads file and generate net from hvm-lang syntax
+fn load_from_lang(file: &str, size: usize) -> (Term, DefNames, hvm_lang::RunInfo) {
+  let path = format!("{}/tests/programs/{}", env!("CARGO_MANIFEST_DIR"), file);
+  let code = fs::read_to_string(path).unwrap();
+
+  let book = parser::parse_definition_book(&code).unwrap();
+  hvm_lang::run_book(book, size).unwrap()
+}
+
+fn total_rewrites(net: &run::Net) -> usize {
```

Comment:
net has a `.rewrites()` method that you can call, no?

Reply:
Yes, I missed that